core/src/test/scala/org/bykn/bosatsu/MatchlessTests.scala

```diff
@@ -151,4 +156,21 @@ class MatchlessTest extends AnyFunSuite {
       assert(matchlessRes == matchRes)
     }
   }
+
+  test("If.flatten can be unflattened") {
```

Comment:
this test didn't always find the bug (since randomly finding a case is somewhat rare, but after 3 runs, one found it, so I think it would be flakey enough that we would discover the issue).